src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+        const handle = options.handle;
+        const service = options.service;
+
+        const successful = await client.findSocialPeer(nodePath, handle, service);
+
+        if (successful) {
+          pkLogger('peer successfully pinged', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('ping timed out', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeStealthCommand() {
+  return new commander.Command('stealth')
+    .description('toggle stealth mode on or off')
+    .option('--node-path <nodePath>', 'node path')
+    .option('--active', 'toggle stealth mode to active')
+    .option('--no-active', 'toggle stealth mode to not active')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const active = options.active;
+        if (active == undefined) {
+          throw Error('active status must be specified with either --active or --no-active');
+        }
+
+        const successful = await client.toggleStealth(nodePath, active);
+
+        pkLogger(`stealth mode toggled to '${active ? 'active' : 'inactive'}'`, PKMessageType.SUCCESS);
+      }),
+    );
+}
+
+function makeRelayCommand() {
+  return new commander.Command('relay')
+    .description('request a relay connection from a public peer')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+
+        const publicKeyPath = options.publicKey;
+        // read in publicKey if it exists
+        const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+        const successful = await client.requestRelay(nodePath, publicKey);
+
+        if (successful) {
+          pkLogger('peer server successfully relayed', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makePunchCommand() {
+  return new commander.Command('punch')
+    .description('request a udp hole punched address from a peer')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
```

Comment:
Yea if you can have common options, this would reduce code duplication.

There are 2 ways:

```
pk [option] subcommand
```

```
pk subcommand [option]
```

In Python argparse the former and latter are separate declarations.

However in our case we can do something similar. And you can always share code that adds options to pre or post. Not sure about commander. Can you comment on this and provide some examples?